app/src/main/java/me/quadphase/qpdex/AdvancedSearch.java

```diff
@@ -0,0 +1,508 @@
+    EditText defGreaterTextUI;
+    EditText spattLowerTextUI;
+    EditText spattGreaterTextUI;
+    EditText spdefLowerTextUI;
+    EditText spdefGreaterTextUI;
+    EditText speedLowerTextUI;
+    EditText speedGreaterTextUI;
+
+    @Override
+    protected void onCreate(Bundle savedInstanceState) {
+        long startTime = System.nanoTime();
+
+        super.onCreate(savedInstanceState);
+        setContentView(R.layout.activity_advanced_search);
+
+        pokemonFactory = PokemonFactory.getPokemonFactory(this.getApplicationContext());
+
+        allTypes = pokemonFactory.getListOfTypes();
+        allAbilities = pokemonFactory.getAllAbilities();
+        allEggGroups = pokemonFactory.getAllEggGroups();
+        allGenerations = pokemonFactory.getGenerations();
+
+        long endTime = System.nanoTime();
+        long duration = (endTime - startTime);
+
+        Log.d("QPDex", String.format("Obtaining Spinner Data took %s ns",duration));
+
+        // No need for MinimalPokemon
+        //allMinimalPokemon = pokemonFactory.getAllMinimalPokemon();
+
+        //setup and fill in all spinner contents
+        retrieveInterfaceElements();
+        fillInSpinnerValues();
+
+        // Provide a set of solutions if no options are selected
+        resetFilter();
+
+        endTime = System.nanoTime();
+        duration = (endTime - startTime);
+
+        Log.d("QPDex", String.format("Total creation took %s ns",duration));
+
+    }
+
+    private void retrieveInterfaceElements(){
+        //Retrieve all UI Variables for setup
+        type1Spinner = (Spinner) findViewById(R.id.type1_spinner);
+        type2Spinner = (Spinner) findViewById(R.id.type2_spinner);
+        abilitySpinner = (Spinner) findViewById(R.id.ability_spinner);
+        eggGroup1Spinner = (Spinner) findViewById(R.id.egg1_spinner);
+        eggGroup2Spinner = (Spinner) findViewById(R.id.egg2_spinner);
+        generationSpinner = (Spinner) findViewById(R.id.generation_spinner);
+
+        hpLowerTextUI = (EditText) findViewById(R.id.hp_lower);
+        hpGreaterTextUI = (EditText) findViewById(R.id.hp_greater);
+        attLowerTextUI = (EditText) findViewById(R.id.att_lower);
+        attGreaterTextUI = (EditText) findViewById(R.id.att_greater);
+        defLowerTextUI = (EditText) findViewById(R.id.def_lower);
+        defGreaterTextUI = (EditText) findViewById(R.id.def_greater);
+        spattLowerTextUI = (EditText) findViewById(R.id.spatt_lower);
+        spattGreaterTextUI = (EditText) findViewById(R.id.spatt_greater);
+        spdefLowerTextUI = (EditText) findViewById(R.id.spdef_lower);
+        spdefGreaterTextUI = (EditText) findViewById(R.id.spdef_greater);
+        speedLowerTextUI = (EditText) findViewById(R.id.speed_lower);
+        speedGreaterTextUI = (EditText) findViewById(R.id.speed_greater);
+
+
+    }
+
+    private void fillInSpinnerValues(){
+
+        //TODO: Filling out these Spinners with the actual objects seems to be causing a bottleneck,

```

Comment:
Removing this comment, the actual bottleneck was at the DB access level. Filling in the spinners is actually quite fast.